olm-catalog/serverless-operator-index/image_content_source_policy.yaml

```diff
@@ -76,84 +62,80 @@ spec:
-      source: "registry.redhat.io/openshift-serverless-1/knative-rhel8-operator"
+      source: "registry.redhat.io/openshift-serverless-1/kn-serving-webhook-rhel8"
     - mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/kourier-control"]
       source: "registry.redhat.io/openshift-serverless-1/kourier-control-rhel8"
-    - mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/net-istio-controller"]
-      source: "registry.redhat.io/openshift-serverless-1/net-istio-controller"
     - mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/net-istio-controller"]
       source: "registry.redhat.io/openshift-serverless-1/net-istio-controller-rhel8"
-    - mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/net-istio-webhook"]
-      source: "registry.redhat.io/openshift-serverless-1/net-istio-webhook"
     - mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/net-istio-webhook"]
       source: "registry.redhat.io/openshift-serverless-1/net-istio-webhook-rhel8"
     - mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/net-kourier-kourier"]
-      source: "registry.redhat.io/openshift-serverless-1/net-kourier-kourier"
-    - mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/serverless-bundle"]
-      source: "registry.redhat.io/openshift-serverless-1/serverless-bundle"
+      source: "registry.redhat.io/openshift-serverless-1/net-kourier-kourier-rhel8"
     - mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/serverless-ingress"]
-      source: "registry.redhat.io/openshift-serverless-1/serverless-ingress"
+      source: "registry.redhat.io/openshift-serverless-1/serverless-ingress-rhel8"
     - mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/serverless-kn-operator"]
-      source: "registry.redhat.io/openshift-serverless-1/serverless-kn-operator"
+      source: "registry.redhat.io/openshift-serverless-1/serverless-kn-operator-rhel8"
     - mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/serverless-openshift-kn-operator"]
-      source: "registry.redhat.io/openshift-serverless-1/serverless-openshift-kn-operator"
+      source: "registry.redhat.io/openshift-serverless-1/serverless-openshift-kn-rhel8-operator"
     - mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/serverless-operator-bundle"]
       source: "registry.redhat.io/openshift-serverless-1/serverless-operator-bundle"
+    - mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/serverless-operator-bundle"]
+      source: "registry.redhat.io/openshift-serverless-1/serverless-operator-bundle-rhel8"
```

Comment:
I suppose this couple of lines will be fixed in the next round of "make generated files"?
We probably want this in the end:
```
- mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/serverless-bundle"]
  source: "registry.redhat.io/openshift-serverless-1/serverless-operator-bundle"
```

Reply:
that one is already there on line 175?

Reply:
I thought this came, as we have in the current version of the catalog (from where this gets created) somewhere the `serverless-operator-bundle-rhel8` around. But as we have
https://github.com/openshift-knative/serverless-operator/blob/ed470411e0a405feaaa388d967203993c6bb91a0/olm-catalog/serverless-operator-index/image_content_source_policy.yaml#L175-L176
I guess we should be fine for this round?

Reply:
Alright. I can see that it's already there on line 175. Thanks.
It's strange that we also have this now:
```
- mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/serverless-operator-bundle"]
  source: "registry.redhat.io/openshift-serverless-1/serverless-operator-bundle"
```
Now sure how this gets there but maybe it will also be removed in the next round?

Reply:
OK. Apparently, we add the correct mirror manually [here](https://github.com/openshift-knative/serverless-operator/blob/main/hack/lib/catalogsource.bash#L163)
And the other wrong mirrors are added automatically by "oc adm catalog mirror":
```
- mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/serverless-operator-bundle"]
  source: "registry.redhat.io/openshift-serverless-1/serverless-operator-bundle"
- mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/serverless-operator-bundle"]
  source: "registry.redhat.io/openshift-serverless-1/serverless-operator-bundle-rhel8"
```
Maybe we can simply drop the manual addition and in the resulting catalog, replace
```
- mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/serverless-operator-bundle"]
```
with
```
- mirrors: ["quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-135/serverless-bundle"]
```
?
It would be easier if the source and target repo were same.

Reply:
Something like this should fix it:
https://gist.github.com/mgencur/fcdf2a81933a6fbfdb421f115614051b

Reply:
I added 04d3b82df64853ea1f64683af462376afd79e46b, in which I removed the manual addition and added a "special case" for the bundle image (similar to what we have for the serverless-openshift-kn-operator name).
This keeps (for now) the `serverless-operator-bundle-rhel8` in the ICSP, but this should be cleaned up, after we don't have it not in the catalog referenced anymore (after this PR got merged).

Reply:
OK. I see you came with something better.